docs/proposals/26-workload-identity.md

```diff
@@ -0,0 +1,547 @@
+allowing service provider specific configurations. Usually, the clients for
+services supporting identity federation need additional information about the
+cloud account and the federated identity in order to successfully use the JWT.
+This information is known to the cloud account owners and they will provide it
+via this `providerConfig` field, for example when AWS is the external service
+the AWS IAM Role ARN will be set in the `providerConfig` field.
+
+The value of the `sub` claim of the OIDC token will be computed by Gardener, it
+will have the following format
+`gardener.cloud:workloadidentity:<workloadidentity-namespace-name>:<workloadidentity-name>:<workloadidentity-uuid>`.
+A validation must ensure that the `WorkloadIdentity` name and namespace name do
+not exceed certain limit. This restrictions is required as per the OIDC
+[Specification](https://openid.net/specs/openid-connect-core-1_0.html#IDToken)
+the `sub` claim cannot exceed 255 ASCII chars length. Gardener API server will
+write the value of the sub claim in the `status.sub` field to make it explicit,
+otherwise Gardener users will have to deduce it themselves which could turn out
+to be error prone.
+
+```yaml
+apiVersion: authentication.gardener.cloud/v1alpha1
+kind: WorkloadIdentity
+metadata:
+  name: banana-testing
+  namespace: garden-local
+  uid: 12b580fe-1f74-4195-852b-e1a74b03496a # generated by the API server.
+spec:
+  audiences: # Required field.
+  - team-foo
+  duration: 48h # Optional field, gardener will have default value of token duration if the field is unset.
+  targetSystem: # Required field.
+    type: aws # Required field.
+    providerConfig: # Optional field of type []byte, extensions can make it mandatory via admission webhooks.
+      apiVersion: aws.authentication.gardener.cloud/v1alpha1
+      kind: Config
+      iamRoleARN: arn:aws:iam::112233445566:role/gardener-dev
+status:
+  sub: gardener.cloud:workloadidentity:garden-local:banana-testing:12b580fe-1f74-4195-852b-e1a74b03496a
+```
+
+JWTs will be available when the clients send `create` requests on the
+`WorkloadIdentity/token` subresource. As the clients will be providing various
+custom information that will be used for the generation of the JWT, yet another
+resource `TokenRequest` in the API group `authentication.gardener.cloud` will be
+used, similar to `TokenRequest` from `authentication.k8s.io/v1` API. It is
+envisioned this resource to contain just metadata for the context where the JWT
+is being used, e.g. shoot or backup entry identifier. Gardener API server must
+verify the provided metadata and it can enhance the JWT with additional
+information derived from the context, for example with information for the
+project and the seed of the shoot cluster. Gardener API can also add global
+information like a garden cluster identity. As `WorkloadIdentity` already
+specifies duration of the token, `TokenRequest` will not feature such field.
```

Comment:
Instead of having this on `WorkloadIdentity`, why not moving the duration to the `TokenRequest` API, similar to how its done for `ServiceAccount`s? You also don't specify the token duration globally on `ServiceAccount` level.

Reply:
Considering the fact that these tokens will be used by Gardener service processes, there is no need to provide any configuration option to the Gardener service consumers (project owners/admins) for defining the validity of these tokens on WorkloadIdentity level. I believe that the validity of these tokens must be defined centrally for the service by the Gardener service operators in a way that it this validity will serve properly all Gardener service processes that will use them and considering common security best practices.

Reply:
The reasons for this design were shared in https://github.com/gardener/gardener/pull/9260#discussion_r1513142814, should I include this also in the GEP?